### PR TITLE
Shadcn sidebar

### PR DESCRIPTION
**Pull request by rohitverma007**
Ready for review.

Migrated the existing navbar to shadcn sidebar component. 

Here's a quick demo video:

https://www.loom.com/share/a36f71ddd540476bbf447efcc4bad8ca?sid=9bf794ec-e8f5-4e91-81fd-7cb65f414fc6

Here are some screenshots:
<details>
<summary>Desktop Screnshots</summary>
Screenshot 2025-04-14 at 5 19 59 PM

<details>
<summary>Mainly would like to get feedback on the company switching logic and if this looks alright:</summary>
Screenshot 2025-04-14 at 5 29 55 PM
Screenshot 2025-04-14 at 5 41 58 PM
</details>

~~The circle notification icon beside company name in dropdown is potentially misleading it's simply meant to be an indication of which company is currently selected. Perhaps a better indication would be to grey out the selected company. ~~ Leaving the default circle icon for now.
</details>

<details>

<summary>The mobile sidebar is a bit different than previous one:</summary
Screenshot 2025-04-17 at 3 29 53 PM
Screenshot 2025-04-17 at 3 29 44 PM
Screenshot 2025-04-17 at 3 29 40 PM

</details>

Todos:

- [x] figure out + test all "company" scenarios - none , 1 , 1+, implement edge cases
- [x] figure out + test all "roles" + implement edge case
- [x] add mobile shadcn sidebar
- [x] refactor SidebarMenuItem (DRYify)
- [ ] fix css to match old nav [leaving this for a different (internal?) PR/ design review ]:
  - [ ]     fix color of default + selected nav item text (match the old navbar)
  - [ ]     fix account icon (default + selected)

part of #18 (and eventually #17)

### CODE REVIEW

**Comment by slavingia** [comment]
Grey out would work. X uses a checkmark, which may work too. But imo it's not bad atm and we can tweak in a future design-minded PR

image

**Comment by slavingia** [comment]
I think going for pixel parity isn't the goal here btw. Shadcn makes some nice improvements to modernize the look, which I tried to copy here - https://github.com/antiwork/flexile/issues/17

But also something we can iterate on internally

**Comment by rohitverma007** [comment]
> I think going for pixel parity isn't the goal here btw. Shadcn makes some nice improvements to modernize the look, which I tried to copy here - #17
> 
> But also something we can iterate on internally

Thanks for the feedback, good to know, will focus less on pixel perfectness. Btw, images on #17 are broken if you are referring to those, can still access v0 fwiw.

**Comment by slavingia** [comment]
Updated them

**Comment by rohitverma007** [comment]
@slavingia wrapped up the remaining items and added a short loom demo video. The PR is ready for review by the team :)
I believe I’ve addressed everything and confident with the testing I have done, but since this touches the app’s main entrypoint, would be worthwhile to give it a thorough review. Thanks!

File: apps/next/components/layouts/Main.tsx [hunk 1]
```diff
@@ -131,206 +150,370 @@ export default function MainLayout({
...
-              </button>
-            </SignOutButton>
+                <SidebarNavItem
+                  href="/settings"
+                  icon={UserIcon}
+                  activeIcon={SolidUserIcon}
+                  isActive={pathname.startsWith("/settings")}
+                  label="Account"
+                />
+                {/* <SidebarMenuItem>
```
Comment (slavingia) [inline apps/next/components/layouts/Main.tsx:360]: ?

**Review by slavingia — CHANGES_REQUESTED** [review]

File: apps/next/components/layouts/Main.tsx (hunk 1 above)
Comment (rohitverma007) [reply apps/next/components/layouts/Main.tsx]: removed 😅

**Comment by slavingia** [comment]
There seem to be other diffs unrelated to this. I've run autofix and the Vercel deploy. Please self-review and then I'll have someone internally do a pass and merge.

**Comment by rohitverma007** [comment]
> There seem to be other diffs unrelated to this. I've run autofix and the Vercel deploy. Please self-review and then I'll have someone internally do a pass and merge.

Indeed, updated to remove the unnecessary changes.

File: apps/next/components/layouts/Main.tsx [hunk 2]
```diff
@@ -364,225 +539,44 @@ const useSwitchCompanyOrRole = () => {
   };
 };
 
-const NavLinks = ({ company }: { company: Company }) => {
```
Comment (raulpopadineti) [inline apps/next/components/layouts/Main.tsx:367]: Why couldn't we have reused this since it's scoped per `company` and makes rendering sidebar nav items cleaner instead of always having to pass the `openCompany.routes.some(route => route.label === label)` for each, like it is now?

Right now it looks like a big diff and hard to determine if something else changed besides replacing the nav item components with Shadcn's.

**Comment by slavingia** [comment]
Great. Turned out to be smaller change than expected which is great. We can merge it in today!

File: apps/next/components/layouts/Main.tsx (hunk 2 above)
Comment (rohitverma007) [reply apps/next/components/layouts/Main.tsx:367]: Ah good point, when I started building it out, it made sense to rewrite all of that but I see the benefit of re-using the NavLinks. Can refactor this, although won't be able to get to it until around the end of the weekend.

Comment (slavingia) [reply apps/next/components/layouts/Main.tsx:367]: Sounds good. Please have done by end of weekend so we can merge in around Noon ET latest on Monday.

Comment (rohitverma007) [reply apps/next/components/layouts/Main.tsx]: @raulpopadineti I have updated the code to re-use the NavLinks and NavLink component which makes the diff more readable. Feel free to give it a look. Reverting back to these components also fixed the other updates href bug. Let me know if there are any other changes I should make, thanks!

File: apps/next/components/layouts/Main.tsx [hunk 3]
```diff
@@ -131,206 +150,362 @@ export default function MainLayout({
...
+        </SidebarHeader>
+        <SidebarContent className="bg-black text-white">
+          {openCompany && (
+            <SidebarGroup>
+              <SidebarGroupContent>
+                <SidebarMenu className="space-y-1">
+                  {openCompany.routes.some(route => route.label === "Updates") && (
+                    <>
+                      <SidebarNavItem
+                        href="/updates"
```
[the remark this answers is not in the document]
Comment (rohitverma007) [reply apps/next/components/layouts/Main.tsx]: Updating to `NavLinks` and `NavLink` fixes this.

**Comment by slavingia** [comment]
Doesn't meet our quality bar. We'll do this internally as it's a small lift.

**Comment by rohitverma007** [comment]
> Doesn't meet our quality bar. We'll do this internally as it's a small lift.

@slavingia For my own understanding and to help me reflect and improve going forward, do you mean the quality in terms of code or the ui/ux? After the re-use of NavLinks and NavLink it seemed like a fairly minor diff and straight forward, any further feedback would be greatly appreciated.

File: apps/next/components/layouts/Main.tsx [hunk 4]
```diff
@@ -36,26 +37,48 @@ import { useQuery, useQueryClient } from "@tanstack/react-query";
...
 import { type Company } from "@/models/user";
 import { trpc } from "@/trpc/client";
-import { cn, e } from "@/utils";
+import { cn } from "@/utils";
 import { assertDefined } from "@/utils/assert";
 import { request } from "@/utils/request";
 import { company_search_path, company_switch_path } from "@/utils/routes";
 import { formatDate } from "@/utils/time";
 import { useOnGlobalEvent } from "@/utils/useOnGlobalEvent";
+import "@/components/layouts/sidebar.css";
```
Comment (slavingia) [inline apps/next/components/layouts/Main.tsx:59]: ?

Comment (rohitverma007) [reply apps/next/components/layouts/Main.tsx:59]: oops, removed.

**Comment by slavingia** [comment]
Sorry for closing the PR with a terse comment. If you want to pair with @MayaRainer or me on re-doing this and merging this in, that may be a better path forward. Just reach out to maya@gumroad.com / sahil@gumroad.com to schedule a call. It's quite urgent so in the next day would be lovely!

**Comment by rohitverma007** [comment]
> Sorry for closing the PR with a terse comment. If you want to pair with @MayaRainer or me on re-doing this and merging this in, that may be a better path forward. Just reach out to [maya@gumroad.com](mailto:maya@gumroad.com) / [sahil@gumroad.com](mailto:sahil@gumroad.com) to schedule a call. It's quite urgent so in the next day would be lovely!

Thanks for this, appreciate the time and help, have reached out to @MayaRainer with some times over the next day. Looking forward to it.

**Comment by slavingia** [comment]
Why isn't this merged?

**Comment by MayaRainer** [comment]
Was on a call with @rohitverma007 and we discussed some changes that had to be made to fix tests and some broken layout. Should be ready for re-review by tomorrow.

**Comment by rohitverma007** [comment]
fixed some things from the call, namely lint issues + using shadcn sidebar for "updates" nav item 
still working through the mobile / defaulting to shadcn sidebar trigger, need a bit more time to wrap that up, should be ready for re-review within a day.

**Comment by MayaRainer** [comment]
Thanks for working on this! Due to the urgency I'm going to bring this over the finish line today, but still feel free to reach out about the bounty when they're implemented.

**Comment by rohitverma007** [comment]
> Thanks for working on this! Due to the urgency I'm going to bring this over the finish line today, but still feel free to reach out about the bounty when they're implemented.

Sure yea, go ahead. Thanks for your help and time. Looking forward to contributing to other issues.

**Comment by slavingia** [comment]
Using a new PR to merge these changes!
